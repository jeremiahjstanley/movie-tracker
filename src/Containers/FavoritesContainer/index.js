import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

const FavoritesContainer = (props) => {
  const displayFavorites = props.favorites.map((favorite, index) => {
    return (
      <Link to={`/favorites/${favorite.title}`} key={`${index} + ${favorite.title}`}>
        <img src={`https://image.tmdb.org/t/p/w600_and_h900_bestv2/${favorite.poster_path}`} width="200px"/>
      </Link>
    );
  });

  return (
    <div>
      { displayFavorites }
    </div>
  );
};


export const mapStateToProps = (state) => ({
  favorites: state.favorites
});


export default connect(mapStateToProps)(FavoritesContainer);