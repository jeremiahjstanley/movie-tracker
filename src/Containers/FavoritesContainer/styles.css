.favorites {
  border: 2px solid chartreuse;
}

.favoritesContainer {
  grid-column: 1;
  display: flex;
  justify-content: space-evenly;
  max-width: 80%;
  overflow: scroll;
}

