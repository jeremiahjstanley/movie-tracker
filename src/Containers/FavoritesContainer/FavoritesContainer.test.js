import React from 'react';
import { shallow } from 'enzyme';
import FavoritesContainer from './';
import { createStore } from 'redux';
import rootReducer from '../../reducers';



describe('Favorites Container Test', () => {
  const store = createStore(rootReducer);

  it('should match the snapshot', () => {
    const wrapper = shallow(<FavoritesContainer store={store}/>);
    
    expect(wrapper).toMatchSnapshot();
  });

});
