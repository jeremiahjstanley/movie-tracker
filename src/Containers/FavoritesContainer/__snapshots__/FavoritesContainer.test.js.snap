// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Container Test should match the snapshot when there are no favorites 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favorites={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "favorites-container",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "favorites-container",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
